tests/appsec/iast/sink/test_weak_hash.py

```diff
@@ -26,29 +25,35 @@
         "*": "?",
     }
 )
-class TestIastWeakHash:
-    """Verify IAST WEAK HASH detection feature"""
-
-    @property
-    def expected_location(self):
-        if context.library.library == "java":
-            return "com.datadoghq.system_tests.iast.utils.CryptoExamples"
-
-        if context.library.library == "nodejs":
-            return "iast.js"
-
-        if context.library.library == "python":
-            # temporary hack because dd-trace-py version is net yet increased
-            is_dev_version = "dev" in str(context.library)
-
-            if context.library.version >= "1.12.0" or is_dev_version:
-                return "iast.py"
-            else:
-                # old value: absolute path
-                if context.weblog_variant == "uwsgi-poc":
-                    return "/app/./iast.py"
-                else:
-                    return "/app/iast.py"
+@missing_feature(context.weblog_variant == "spring-boot-native", reason="GraalVM. Tracing support only")
+@missing_feature(context.weblog_variant == "spring-boot-3-native", reason="GraalVM. Tracing support only")
+class TestWeakHash:
+    """Verify weak hash detection."""
+
+    sink_fixture = SinkFixture(
+        vulnerability_type="WEAK_CIPHER",
+        insecure_endpoint="/iast/insecure_hashing/test_md5_algorithm",
+        secure_endpoint="/iast/insecure_hashing/test_secure_algorithm",
+        data=None,
+        location_map={
+            "java": "com.datadoghq.system_tests.iast.utils.CryptoExamples",
+            "nodejs": "iast.js",
+            "python": "iast.py",
+        },
+        evidence_map="md5",
+    )
+
+    def setup_insecure(self):
+        self.sink_fixture.setup_insecure()
+
+    def test_insecure(self):
+        self.sink_fixture.setup_insecure()
+
+    def setup_secure(self):
+        self.sink_fixture.setup_secure()
+
+    def test_secure(self):
+        self.sink_fixture.test_secure()
 
     def setup_insecure_hash_remove_duplicates(self):
```

Comment:
At some point we will probably want to move this tests to a different place for vulnerability deduplication